apps/mobile/src/navigations/root/root.tsx

```diff
@@ -24,44 +23,36 @@ const screenOptions: NativeStackNavigationOptions = {
 
 const Root: React.FC = () => {
 	const user = useAppSelector((state) => state.auth.user);
+
+	const dispatch = useAppDispatch();
+
+	const hasAnsweredQuizQuestions = user?.hasAnsweredQuizQuestions;
 	const hasUser = Boolean(user);
 
+	const filteredNavigationItems = useMemo(() => {
```

Comment:
Let's maybe create a hook that will return already filtered list of screens.
Then we don't need `NAVIGATION_ITEMS ` constant and we'll be able to replace `checkShouldBeRendered ` callback with a simple boolean `shouldBeRendered`